### PR TITLE
Tgui in-hand material sheet construction menu

### PR DESCRIPTION
**Pull request by Garash2k**
[Feature][QoL][tgui]

## About the PR 
Replaces the old html in-hand material sheet menu with a TGUI one.
Inspired by the human inventory menu and the seed fabricator.
image
image
image

## Why's this needed? 
ICONS!!!!
Having a button become disabled after a craft is good feedback.
Also TGUI looks better and is more responsive.

## Changelog

```changelog
(u)Garash
(+)New UI for the material sheet's in-hand construction menu.
```

### CODE REVIEW

**Comment by Garash2k** [comment]
Wow, there's still weird lingering commits from exactly 4 days ago, I've completely reset my origin/master branch to upstream/master so I imagine it'll be fine next time.

**Comment by ZeWaka** [comment]
It's because you keep rebasing

**Comment by Garash2k** [comment]
Yeah I figured it's related to that one bad merge, but I didn't expect it to be a problem in the long term, should be fine with a new master from now on

File: tgui/packages/tgui/interfaces/SheetCrafting.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+export const SheetCrafting = (_props, context) => {
+  const { data } = useBackend(context);
+  const { availableAmount, labeledAvailableAmount, itemList } = data;
+
+  return (
+    <Window
+      width={360}
+      height={690}>
+      <Window.Content>
+        <Section title={"Recipes"}>
```
Comment (Luxizzle) [inline tgui/packages/tgui/interfaces/SheetCrafting.js:21]: Can you divide these 2 sections with a `Stack` so you can make the first one scrollable and the second one is always visible? 
image

**Review by Luxizzle — CHANGES_REQUESTED** [review]
TGUI code looks good 👍 just a small change in the layout and I'm happy

**Comment by Luxizzle** [comment]
TGUI looks good! Recipes could *probably* be moved into ui_static_data, but it isn't a big deal I think.

**Comment by Garash2k** [comment]
I doubt the "computer3.dm,656: Cannot modify null.plane." error is caused by this PR, any ways to re run the checks? Or is taking no action ok?

**Comment by Garash2k** [comment]
> TGUI looks good! Recipes could _probably_ be moved into ui_static_data, but it isn't a big deal I think.

Thanks for telling me about ui_static_data, but I've actually thought of a reason not to use it!

Say you whack the metal sheet into an artifact Matter Transmutation bomb, it then becomes glass, so it's good that the recipes are eventually gonna update!

**Comment by Tarmunora** [comment]
!merge_upstream

**Comment by Garash2k** [comment]
!merge_upstream

**Comment by BatElite** [comment]
Unless you code in TGUI support for ghostdrones this would effectively break them.

**Comment by Garash2k** [comment]
I think, making use of the `isghostdrone()` function and `NO_GHOSTCRITTER` flag, I've got a reasonable solution.
image

**Comment by Garash2k** [comment]
@ZeWaka can the Stale timer be reset on this pull request? The last outstanding point was addressed and I've tested it post feature freeze.

**Pull request merged**